NitroxLauncher/Models/Utils/WindowsHelper.cs

```diff
@@ -55,6 +64,119 @@ internal static void RestartAsAdmin()
                 Log.Info("Can't restart the launcher as administrator, we already have permissions");
             }
         }
-    }
 
+        internal static void CheckFirewallRules()
+        {
+            try
+            {
+                CheckFirewallRules(FirewallDirection.Inbound);
+                CheckFirewallRules(FirewallDirection.Outbound);
+            }
+            catch (FileNotFoundException ex)
+            {
+                Log.Warn($"Tried to add firewall rule for program that does not exist: {ex.FileName}");
+            }
+            catch (UnauthorizedAccessException)
+            {
+                if (!Settings.Default.FirewallWarningShown)
+                {
+                    Settings.Default.FirewallWarningShown = true;
+                    Settings.Default.Save();
+                    MessageBox.Show("Try restarting the launcher as administrator or manually adding firewall rules for Nitrox programs. This warning won't be shown again.", "Error adding Windows Firewall rules", MessageBoxButton.OK, MessageBoxImage.Warning);
+                }
+            }
+        }
+
+        internal static void CheckFirewallRules(FirewallDirection direction)
+        {
+            CheckClientFirewallRules(direction);
+            CheckServerFirewallRules(direction);
+            CheckHamachiFirewallRules(direction);
+        }
+
+        private static void CheckServerFirewallRules(FirewallDirection direction)
+        {
+            string serverPath = Path.Combine(Path.GetDirectoryName(Assembly.GetEntryAssembly().Location), ServerLogic.SERVER_EXECUTABLE);
+
+            AddExclusiveFirewallRule(Path.GetFileName(serverPath), serverPath, direction);
+        }
+
+        private static void CheckClientFirewallRules(FirewallDirection direction)
+        {
+            string clientPath = Path.Combine(LauncherLogic.Config.SubnauticaPath, "Subnautica.exe");
```

Comment:
We can replace the game exe by the constants defined in `NitroxModel.GameInfo.Subnautica.ExeName`